### ISSUE TITLE
Training stops after 15/16 steps - is it supposed to?

### ISSUE THREAD

**Issue by vijaygill** [tensorpack/tensorpack#1383]
Hi,

Sorry for another question.

The training stops after 15-16 epochs no matter ho big the dataset it. Is it supposed to do so? I do not see anything in the logs. Here is the log of the last epoch.

[0119 15:37:31 @base.py:273] Start Epoch 16 ...
100%|##########################################################################################################################|500/500[01:49<00:00, 4.55it/s]
[0119 15:39:21 @base.py:283] Epoch 16 (global_step 8000) finished, time:1 minute 49 seconds.
[0119 15:39:22 @saver.py:82] Model saved to /training/logs/gillsoft/model-8000.
[0119 15:39:22 @monitor.py:474] GPUUtil/0: 94.606
[0119 15:39:22 @monitor.py:474] HostFreeMemory (GB): 52.046
[0119 15:39:22 @monitor.py:474] PeakMemory(MB)/gpu:0: 3799.9
[0119 15:39:22 @monitor.py:474] QueueInput/queue_size: 50
[0119 15:39:22 @monitor.py:474] Throughput (samples/sec): 4.5542
[0119 15:39:22 @monitor.py:474] fastrcnn_losses/box_loss: 0.12002
[0119 15:39:22 @monitor.py:474] fastrcnn_losses/label_loss: 0.14164
[0119 15:39:22 @monitor.py:474] fastrcnn_losses/label_metrics/accuracy: 0.95489
[0119 15:39:22 @monitor.py:474] fastrcnn_losses/label_metrics/false_negative: 0.409
[0119 15:39:22 @monitor.py:474] fastrcnn_losses/label_metrics/fg_accuracy: 0.49275
[0119 15:39:22 @monitor.py:474] fastrcnn_losses/num_fg_label: 33.171
[0119 15:39:22 @monitor.py:474] learning_rate: 0.001
[0119 15:39:22 @monitor.py:474] maskrcnn_loss/accuracy: 0.89316
[0119 15:39:22 @monitor.py:474] maskrcnn_loss/fg_pixel_ratio: 0.50503
[0119 15:39:22 @monitor.py:474] maskrcnn_loss/maskrcnn_loss: 0.2414
[0119 15:39:22 @monitor.py:474] maskrcnn_loss/pos_accuracy: 0.45229
[0119 15:39:22 @monitor.py:474] mean_gt_box_area: 1.5406e+05
[0119 15:39:22 @monitor.py:474] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level2: 241.36
[0119 15:39:22 @monitor.py:474] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level3: 146.33
[0119 15:39:22 @monitor.py:474] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level4: 101.74
[0119 15:39:22 @monitor.py:474] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level5: 22.568
[0119 15:39:22 @monitor.py:474] multilevel_roi_align_mask/fpn_map_rois_to_levels/num_roi_level2: 1.9183
[0119 15:39:22 @monitor.py:474] multilevel_roi_align_mask/fpn_map_rois_to_levels/num_roi_level3: 8.2557
[0119 15:39:22 @monitor.py:474] multilevel_roi_align_mask/fpn_map_rois_to_levels/num_roi_level4: 17.116
[0119 15:39:22 @monitor.py:474] multilevel_roi_align_mask/fpn_map_rois_to_levels/num_roi_level5: 5.8814
[0119 15:39:22 @monitor.py:474] rpn_losses/box_loss: 0.0057557
[0119 15:39:22 @monitor.py:474] rpn_losses/label_loss: 0.016183
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/box_loss: 2.0341e-06
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/label_loss: 0.00020969
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/label_metrics/precision_th0.1: 0.43959
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/label_metrics/precision_th0.2: 0.4756
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/label_metrics/precision_th0.5: 0.50002
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/label_metrics/recall_th0.1: 0.49986
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/label_metrics/recall_th0.2: 0.49986
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/label_metrics/recall_th0.5: 0.49985
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/num_pos_anchor: 0.0014757
[0119 15:39:22 @monitor.py:474] rpn_losses/level2/num_valid_anchor: 194.14
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/box_loss: 0.00025257
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/label_loss: 0.0013745
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/label_metrics/precision_th0.1: 0.42515
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/label_metrics/precision_th0.2: 0.43369
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/label_metrics/precision_th0.5: 0.51689
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/label_metrics/recall_th0.1: 0.54533
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/label_metrics/recall_th0.2: 0.54517
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/label_metrics/recall_th0.5: 0.53847
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/num_pos_anchor: 0.39118
[0119 15:39:22 @monitor.py:474] rpn_losses/level3/num_valid_anchor: 44.638
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/box_loss: 0.0014121
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/label_loss: 0.010028
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/label_metrics/precision_th0.1: 0.47898
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/label_metrics/precision_th0.2: 0.49385
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/label_metrics/precision_th0.5: 0.56775
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/label_metrics/recall_th0.1: 0.57506
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/label_metrics/recall_th0.2: 0.5562
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/label_metrics/recall_th0.5: 0.5416
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/num_pos_anchor: 1.7031
[0119 15:39:22 @monitor.py:474] rpn_losses/level4/num_valid_anchor: 10.661
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/box_loss: 0.0018185
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/label_loss: 0.0033942
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/label_metrics/precision_th0.1: 0.6399
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/label_metrics/precision_th0.2: 0.64469
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/label_metrics/precision_th0.5: 0.67431
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/label_metrics/recall_th0.1: 0.74208
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/label_metrics/recall_th0.2: 0.73539
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/label_metrics/recall_th0.5: 0.73054
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/num_pos_anchor: 3.0655
[0119 15:39:22 @monitor.py:474] rpn_losses/level5/num_valid_anchor: 4.5066
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/box_loss: 0.0022704
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/label_loss: 0.0011763
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/label_metrics/precision_th0.1: 0.72632
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/label_metrics/precision_th0.2: 0.72635
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/label_metrics/precision_th0.5: 0.73373
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/label_metrics/recall_th0.1: 0.74797
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/label_metrics/recall_th0.2: 0.73968
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/label_metrics/recall_th0.5: 0.73968
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/num_pos_anchor: 1.9633
[0119 15:39:22 @monitor.py:474] rpn_losses/level6/num_valid_anchor: 2.05
[0119 15:39:22 @monitor.py:474] sample_fast_rcnn_targets/num_bg: 478.83
[0119 15:39:22 @monitor.py:474] sample_fast_rcnn_targets/num_fg: 33.171
[0119 15:39:22 @monitor.py:474] sample_fast_rcnn_targets/proposal_metrics/best_iou_per_gt: 0.82149
[0119 15:39:22 @monitor.py:474] sample_fast_rcnn_targets/proposal_metrics/recall_iou0.3: 1
[0119 15:39:22 @monitor.py:474] sample_fast_rcnn_targets/proposal_metrics/recall_iou0.5: 1
[0119 15:39:22 @monitor.py:474] total_cost: 0.71978
[0119 15:39:22 @monitor.py:474] wd_cost: 0.19478
[0119 15:39:22 @base.py:287] Training has finished!
2020-01-19 15:39:22.282170: W tensorflow/core/kernels/queue_base.cc:277] _0_QueueInput/input_queue: Skipping cancelled enqueue attempt with queue not closed
MultiProcessMapDataZMQ successfully cleaned-up.

**Comment by ppwwyyxx** [comment]
You specify how long it trains with the `LR_SCHEDULE` option.

**Comment by vijaygill** [comment]
> You specify how long it trains with the `LR_SCHEDULE` option.

@ppwwyyxx Thank you once again! I just copied the command-line from balloon example and did not attempt to read comments in config.py (a bit of hurry and excitement to get things going).

I will give it a go and report here.

**Comment by vijaygill** [comment]
@ppwwyyxx Just to report that now my training has gone beyond the number of epochs where it used to stop. Thanks again for your quick response!
